### PR TITLE
Fix non-nullable datetime when zeroDateTimeBehavior is CONVERT_TO_NULL.

### PR DESCRIPTION
**Pull request by damjad**
It's for the issue https://github.com/data-integrations/database-plugins/issues/488 

The fix contains a check if the `zeroDateTimeBehavior` is to convert to null, make all date time like columns nullable.

### CODE REVIEW

**Comment by damjad** [comment]
Hello @itsankit-google,

Is there anything more that needs to be done?

**Comment by damjad** [comment]
Hello @itsankit-google,

Could you please have a look? 
Could you please guide me if there's anything more that should be done? 

We need this patch when we upgrade to the latest version of data fusion. Thanks!

**Review by itsankit-google — COMMENTED** [review]
Please add unit tests.

**Comment by damjad** [comment]
I have added the tests and run them.

I ran the following command:

```
mvn test -Dtest="MysqlSchemaReaderUnitTest,MysqlUtilUnitTest"
```

The output is as follows:

```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running io.cdap.plugin.mysql.MysqlSchemaReaderUnitTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.814 s - in io.cdap.plugin.mysql.MysqlSchemaReaderUnitTest
[INFO] validateZeroDateTimeBehavior(io.cdap.plugin.mysql.MysqlSchemaReaderUnitTest)  Time elapsed: 0.741 s
[INFO] validateYearTypeToStringTypeConversion(io.cdap.plugin.mysql.MysqlSchemaReaderUnitTest)  Time elapsed: 0.007 s
[INFO] Running io.cdap.plugin.mysql.MysqlUtilUnitTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.064 s - in io.cdap.plugin.mysql.MysqlUtilUnitTest
[INFO] testIsZeroDateTimeToNull(io.cdap.plugin.mysql.MysqlUtilUnitTest)  Time elapsed: 0.004 s
[INFO] testIsDateTimeLikeType(io.cdap.plugin.mysql.MysqlUtilUnitTest)  Time elapsed: 0 s
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0

```

File: mysql-plugin/src/main/java/io/cdap/plugin/mysql/MysqlSchemaReader.java [hunk 1]
```diff
@@ -31,12 +35,43 @@ public class MysqlSchemaReader extends CommonSchemaReader {
   public static final String YEAR_TYPE_NAME = "YEAR";
   public static final String MEDIUMINT_UNSIGNED_TYPE_NAME = "MEDIUMINT UNSIGNED";
   private final String sessionID;
+  private boolean zeroDateTimeToNull;
+
```
Comment (itsankit-google) [inline mysql-plugin/src/main/java/io/cdap/plugin/mysql/MysqlSchemaReader.java:39]: nit: remove extra empty line.

File: mysql-plugin/src/main/java/io/cdap/plugin/mysql/MysqlSource.java [hunk 2]
```diff
@@ -81,7 +81,7 @@ protected LineageRecorder getLineageRecorder(BatchSourceContext context) {
 
   @Override
   protected SchemaReader getSchemaReader() {
-    return new MysqlSchemaReader(null);
+    return new MysqlSchemaReader(null, mysqlSourceConfig.getConnectionArguments());
```
Comment (itsankit-google) [inline mysql-plugin/src/main/java/io/cdap/plugin/mysql/MysqlSource.java:84]: This change is also needed here in `MysqlConnector`: https://github.com/data-integrations/database-plugins/blob/9ad84e9f50db276d87ce0f6a250576ac4998f4c7/mysql-plugin/src/main/java/io/cdap/plugin/mysql/MysqlConnector.java#L65

**Review by itsankit-google — COMMENTED** [review]
Please also make the similar changes in `CloudSQLMySQL` plugin for consistency.

**Review by itsankit-google — APPROVED** [review]

**Comment by itsankit-google** [comment]
Please squash commits before we merge the PR, you can refer to steps here: https://github.com/data-integrations/firestore-plugins/pull/18#issuecomment-1965094961

**Comment by damjad** [comment]
@itsankit-google squashed 🎊

File: mysql-plugin/src/test/java/io/cdap/plugin/mysql/MysqlUtilUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    int timestampWithTimezoneType = Types.TIMESTAMP_WITH_TIMEZONE;
+    int timeType = Types.TIME;
+    int stringType = Types.VARCHAR;
+
+    assertTrue(MysqlUtil.isDateTimeLikeType(dateType));
+    assertTrue(MysqlUtil.isDateTimeLikeType(timestampType));
+    assertTrue(MysqlUtil.isDateTimeLikeType(timestampWithTimezoneType));
+    assertFalse(MysqlUtil.isDateTimeLikeType(timeType));
+    assertFalse(MysqlUtil.isDateTimeLikeType(stringType));
+
```
Comment (itsankit-google) [inline mysql-plugin/src/test/java/io/cdap/plugin/mysql/MysqlUtilUnitTest.java]: one nit: please remove extra empty line.

Sorry the late comment.

**Pull request merged**